### ISSUE TITLE
Not able build the project , it is constantly showing errors on after the other. 

### ISSUE THREAD

**Issue by shivanshtech** [aws/aws-sdk-cpp#1537]
Not able to build third party dependencies properly.
Not able to hence build the project. I'm a fresher , is there a step-by-step justification of the steps.

I followed the steps and the error are like 

  Performing configure step for 'AwsCCommon'
  -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
  -- Building SIMD base64 decoder
  -- Packaging is only supported on Linux
  -- Configuring done
CUSTOMBUILD : CMake error : install(EXPORT "aws-c-common-targets" ...) includes target "aws-c-common" more than once in the export set. [C:\abc\.deps\AwsCCommon.vcxproj]
  -- Generating done
  CMake Generate step failed.  Build files cannot be regenerated correctly.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(238,5): error MSB8066: Custom build for 'C:\abc\.deps\CMakeFiles\2bcb0194e3f055634582c0f49f30de41\AwsCCommon-configure.rule;C:\abc\.deps\CMakeFiles\2bcb0194e3f055634582c0f49f30de41\AwsCCommon-build.rule;C:\abc\.deps\CMakeFiles\2bcb0194e3f055634582c0f49f30de41\AwsCCommon-install.rule;C:\abc\.deps\CMakeFiles\132c32d64f9eef760aa8b2e77224d2c3\AwsCCommon-complete.rule;C:\abc\.deps\CMakeFiles\d89dda79997f284cad67bdb0c2a161f3\AwsCCommon.rule' exited with code 1. [C:\abc\.deps\AwsCCommon.vcxproj]
  Performing configure step for 'AwsChecksums'
  -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
  -- Configuring done
CUSTOMBUILD : CMake error : install(EXPORT "aws-checksums-targets" ...) includes target "aws-checksums" more than once in the export set. [C:\abc\.deps\AwsChecksums.vcxproj]
  -- Generating done
  CMake Generate step failed.  Build files cannot be regenerated correctly.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(238,5): error MSB8066: Custom build for 'C:\abc\.deps\CMakeFiles\cde3193762586c03baed1ee0bae72ecc\AwsChecksums-configure.rule;C:\abc\.deps\CMakeFiles\cde3193762586c03baed1ee0bae72ecc\AwsChecksums-build.rule;C:\abc\.deps\CMakeFiles\cde3193762586c03baed1ee0bae72ecc\AwsChecksums-install.rule;C:\abc\.deps\CMakeFiles\132c32d64f9eef760aa8b2e77224d2c3\AwsChecksums-complete.rule;C:\abc\.deps\CMakeFiles\d89dda79997f284cad67bdb0c2a161f3\AwsChecksums.rule' exited with code 1. [C:\abc\.deps\AwsChecksums.vcxproj]
CMake Error at CMakeLists.txt:224 (message):
  Failed to build third-party libraries.

-- Configuring incomplete, errors occurred!

Later ALLBUILD HAS ERRORS LIKE Severity	Code	Description	Project	File	Line	Suppression State
Error		install(EXPORT "aws-checksums-targets" ...) includes target "aws-checksums" more than once in the export set.	AwsChecksums	\.deps\CUSTOMBUILD	1	

Confirm by changing [ ] to [x] below:
- [X ] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) and [API reference](http://sdk.amazonaws.com/cpp/api/LATEST/index.html)
- [X ] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-cpp/issues) and didn't find any solution

**Platform/OS/Hardware/Device** WINDOWS 10
What are you running the sdk on?

**Describe the question**

**Logs/output**
If applicable, add logs or error output.

To enable logging, set the following system properties:

*REMEMBER TO SANITIZE YOUR PERSONAL INFO*

```
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Trace;
Aws::InitAPI(options)
```

**Comment by KaibaLopez** [comment]
Hi @shivanshtech ,

We do have instructions on how to install on the [ReadMe](https://github.com/aws/aws-sdk-cpp#building-from-source). Or you could follow the first part of this [dev blog post](https://aws.amazon.com/blogs/gametech/how-to-build-and-integrate-the-aws-sdk-for-c-games-on-windows/).
But if you are having problems building I may be able to help, can you show me what your cmake command looks like? Also, can you verify that you're not trying to install in a location that needs administrator access?

**Comment by shivanshtech** [comment]
i'm using commands as mentioned but constantly showing errors-
cd C:/project1/
cmake D:\sdk\aws-sdk-cpp\aws-cpp-sdk-secretsmanager -G "Visual Studio 16
2019" -DCMAKE_BUILD_TYPE=Debug
msbuild ALL_BUILD.vcxproj /p:Configuration=Debug

**Comment by shivanshtech** [comment]
[CMakeOutput.log](https://github.com/aws/aws-sdk-cpp/files/5695827/CMakeOutput.log)
 
This is the output log generated after running above mentioned three cmake commands

**Comment by shivanshtech** [comment]
@KaibaLopez kindly look into the matter

**Comment by rankela** [comment]
I have the same issue on Ubuntu 20.06 and cmake 3.19.1.
Cmake command:
`cd Release`
`cmake .. -DCMAKE_BUILD_TYPE=Release`

**Comment by KaibaLopez** [comment]
Hi @rankela  and @shivanshtech ,
Hmm yea ok I was able to reproduce the issue on cmake version 3.19.1.
This will probably take us some time to figure out because it is on one of the dependencies rather than the SDK itself.

For now as a work around, try using an older version of cmake to build, 3.18 should work fine.

**Comment by lizroth** [comment]
@KaibaLopez This appears to be a CMake issue that has since been fixed. We were having the same issue with our CI configuration. CMake 3.19.3 works. I didn't do a deep dive on what exactly happened though or whether 3.19.2 works.

Hope this helps. Cheers

https://github.com/aws/aws-encryption-sdk-c/issues/658

**Comment by boinst** [comment]
I also had this issue on CMake 3.19.0 (Windows, VS2019), and the issue is resolved on CMake 3.19.4.

**Comment by KaibaLopez** [comment]
Alright, I think it's safe to close the issue then, thank you for letting me know.
